draft-ounsworth-pq-composite-sigs.mkd

```diff
@@ -323,6 +341,26 @@ Where each BIT STRING within the SEQUENCE is a signature value produced by one o
 
 The choice of `SEQUENCE OF BIT STRING`, rather than for example a single BIT STRING containing the concatenated signature values, is to gracefully handle variable-length signature values by taking advantage of ASN.1's build-in length fields.
 
+
+### Explicit Composite Signature
+
+The ASN.1 algorithm object for an explicit composite signature is:
+
+~~~ asn.1
+sa-CompositeSignature SIGNATURE-ALGORITHM ::= {
+    IDENTIFIER id-alg-explicit
+    VALUE CompositeSignatureValue
+    PARAMS ARE optional
+    PUBLIC-KEYS { pk-Composite }
+    SMIME-CAPS { IDENTIFIED BY id-alg-explicit } }
+}
+~~~
+
+In this case, id-alg-explicit is an algorithm and OID defined elsewhere which explicitly defines the component algorithms. PARAMS are optional as they will likely carry redundant information to id-alg-explicit. The definition of id-alg-explicit needs to specify the contents of VALUE, PARAMS, and SMIME-CAPS.
+
```

Comment:
This and the definition of explicit composite public keys seems too generic; I think the value here would be making it possible to easily define an explicit OID for a particular pair of algorithms.  If the user is still left to define the syntax of their public keys and signatures, we're not providing much help.  If they want to do their own thing, they can without referencing this draft.

This is new to me too, but I'd suggest using [Information Objects](https://www.oss.com/asn1/resources/asn1-made-simple/advanced-topics.html), so that the user can provide the pair-specific OID, key and parameter syntaxes and then our spec defines how they are used.  Done this way there isn't much that changes going from generic to explicit; you just leave out the two algorithm OIDs in the encodings.  But I think the benefit is that only the specific pair is allowed.

Reply:
Remove the ".. or you can do whatever you want" text.
Let's be more prescriptive and rigidly define our encoding.